### ISSUE TITLE
The conversion augmentation always returns an roi.

### ISSUE THREAD

**Issue by AjeyPaiK** [NKI-AI/dlup#116]
In my loss implementation, I use the ROIs to precisely backpropagate loss contributions from only those pixel regions in the H&E images over which there is an ROI defined containing our classes of interest.

A generic representation of my loss looks like this: 
```
if batch["roi"] is not None: 
    batch_loss = loss_fn(rois*preds, rois*targets)
else:
    batch_loss = loss_fn(preds, targets)
```
If the training batch doesn't contain `roi` (meaning, there are no ground truth labels), then the loss function should not consume rois.

But right now, the [dlup transform](https://github.com/NKI-AI/dlup/blob/e327d6d7996f5d180e2865c21c59464c50d3cdf1/dlup/data/transforms.py#L58) always outputs an empty roi (filled with zeros) even though there is no mask corresponding to any label in a particular tile.

So, the loss curve looks like this:
![image](https://user-images.githubusercontent.com/26798611/195316861-46930873-3ae4-40bd-8b7a-aa74bf963066.png)

Meaning, the loss computation happens only when there is a tile with a label and all other time, my model outputs become zero because they get multiplied by the "roi" every time even though there is no ROI.

I handle this within my own codebase. Although, it would be nice if the transform returned an ROI array only if there is a corresponding label within it else, it should return `None`.

**Comment by jonasteuwen** [comment]
This is expected behaviour. If there is a ROI name and no ROI present the output should be zeros.
